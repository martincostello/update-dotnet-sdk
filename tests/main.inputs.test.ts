// Copyright (c) Martin Costello, 2020. All rights reserved.
// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.

import * as core from '@actions/core';
import { afterAll, beforeAll, describe, expect, jest, test } from '@jest/globals';
import { ActionFixture } from './ActionFixture';

const timeout = 30000;
const outputs = [
  ['aspnetcore-version'],
  ['branch-name'],
  ['pull-request-html-url'],
  ['pull-request-number'],
  ['pull-requests-closed'],
  ['runtime-version'],
  ['sdk-updated'],
  ['sdk-version'],
  ['security'],
  ['windows-desktop-version'],
];

describe('update-dotnet-sdk', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => new Date(Date.UTC(2023, 8 - 1, 25)).valueOf());
  });

  describe.each([
    ['close-superseded', 'false', 'close-superseded-false'],
    ['close-superseded', 'true', 'close-superseded-true'],
    ['dry-run', 'true', 'dry-run'],
  ])('when %s is %s', (name: string, value: string, scenario: string) => {
    let fixture: ActionFixture;

    beforeAll(async () => {
      fixture = new ActionFixture('6.0.100');
      await fixture.initialize(scenario, {
        [name]: value,
      });
    });

    afterAll(async () => {
      await fixture?.destroy();
    });

    describe('running the action', () => {
      beforeAll(async () => {
        await fixture.run();
      }, timeout);

      test('generates no errors', () => {
        expect(core.error).toHaveBeenCalledTimes(0);
        expect(core.setFailed).toHaveBeenCalledTimes(0);
      });

      test('updates the SDK version in global.json', async () => {
        expect(await fixture.sdkVersion()).toMatchSnapshot();
      });

      test('generates the expected Git commit history', async () => {
        expect(await fixture.commitHistory(1)).toMatchSnapshot();
      });

      test.each(outputs)('the %s output is correct', (name: string) => {
        expect(fixture.getOutput(name)).toMatchSnapshot();
      });
    });
  });
});
