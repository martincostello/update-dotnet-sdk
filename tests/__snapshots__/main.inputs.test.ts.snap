// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-dotnet-sdk when close-superseded is false running the action generates the expected Git commit history 1`] = `
[
  "Update .NET SDK",
  "",
  "Update .NET SDK to version 6.0.413.",
  "",
  "---",
  "updated-dependencies:",
  "- dependency-name: Microsoft.NET.Sdk",
  "  dependency-type: direct:production",
  "  update-type: version-update:semver-patch",
  "...",
  "",
  "Signed-off-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>",
]
`;

exports[`update-dotnet-sdk when close-superseded is false running the action the branch-name output is correct 1`] = `"update-dotnet-sdk-6.0.413"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the pull-request-html-url output is correct 1`] = `"https://github.local/martincostello/update-dotnet-sdk/pull/42"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the pull-request-number output is correct 1`] = `"42"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the pull-requests-closed output is correct 1`] = `"[]"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the sdk-updated output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the sdk-version output is correct 1`] = `"6.0.413"`;

exports[`update-dotnet-sdk when close-superseded is false running the action the security output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when close-superseded is false running the action updates the SDK version in global.json 1`] = `"6.0.413"`;

exports[`update-dotnet-sdk when close-superseded is true running the action generates the expected Git commit history 1`] = `
[
  "Update .NET SDK",
  "",
  "Update .NET SDK to version 6.0.413.",
  "",
  "---",
  "updated-dependencies:",
  "- dependency-name: Microsoft.NET.Sdk",
  "  dependency-type: direct:production",
  "  update-type: version-update:semver-patch",
  "...",
  "",
  "Signed-off-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>",
]
`;

exports[`update-dotnet-sdk when close-superseded is true running the action the branch-name output is correct 1`] = `"update-dotnet-sdk-6.0.413"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the pull-request-html-url output is correct 1`] = `"https://github.local/martincostello/update-dotnet-sdk/pull/42"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the pull-request-number output is correct 1`] = `"42"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the pull-requests-closed output is correct 1`] = `"[40,41]"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the sdk-updated output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the sdk-version output is correct 1`] = `"6.0.413"`;

exports[`update-dotnet-sdk when close-superseded is true running the action the security output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when close-superseded is true running the action updates the SDK version in global.json 1`] = `"6.0.413"`;

exports[`update-dotnet-sdk when dry-run is true running the action generates the expected Git commit history 1`] = `
[
  "Update .NET SDK",
  "",
  "Update .NET SDK to version 6.0.413.",
  "",
  "---",
  "updated-dependencies:",
  "- dependency-name: Microsoft.NET.Sdk",
  "  dependency-type: direct:production",
  "  update-type: version-update:semver-patch",
  "...",
  "",
  "Signed-off-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>",
]
`;

exports[`update-dotnet-sdk when dry-run is true running the action the branch-name output is correct 1`] = `""`;

exports[`update-dotnet-sdk when dry-run is true running the action the pull-request-html-url output is correct 1`] = `""`;

exports[`update-dotnet-sdk when dry-run is true running the action the pull-request-number output is correct 1`] = `"0"`;

exports[`update-dotnet-sdk when dry-run is true running the action the pull-requests-closed output is correct 1`] = `"[]"`;

exports[`update-dotnet-sdk when dry-run is true running the action the sdk-updated output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when dry-run is true running the action the sdk-version output is correct 1`] = `"6.0.413"`;

exports[`update-dotnet-sdk when dry-run is true running the action the security output is correct 1`] = `"true"`;

exports[`update-dotnet-sdk when dry-run is true running the action updates the SDK version in global.json 1`] = `"6.0.413"`;
